### PR TITLE
ChannelData: fix padding

### PR DESCRIPTION
**Pull request by champtar**
Padding is mandatory for tcp but optionnal for udp
See https://tools.ietf.org/html/rfc5766#section-11.5

### CODE REVIEW

**Comment by ibauersachs** [comment]
Without looking at the actual padding change, I'm against merging this because it changes (removes) public API.

File: src/main/java/org/ice4j/message/ChannelData.java [hunk 1]
```diff
@@ -196,7 +196,12 @@ public static ChannelData decode(byte binMessage[], char offset, char arrayLen)
         }
 
         msgLen = (char)((binMessage[offset++]<<8) | (binMessage[offset++]&0xFF));
-        if(msgLen != (binMessage.length - 4))
+        /*
+         * message can be padded
+         * https://tools.ietf.org/html/rfc5766#section-11.5
+         */
+        if (msgLen != (binMessage.length - HEADER_LENGTH)
+            && ((msgLen % 4) + 4) != (binMessage.length - HEADER_LENGTH))
```
Comment (ibauersachs) [inline src/main/java/org/ice4j/message/ChannelData.java]: I might be tired and don't know ICE or TURN, but I'd assume that `binMessage.length - HEADER_LENGTH` can be legally greater than `[4,7]`.

Comment (champtar) [inline src/main/java/org/ice4j/message/ChannelData.java]: yep this is wrong, you should read ```msgLen - (msgLen % 4) + 4```

Comment (champtar) [inline src/main/java/org/ice4j/message/ChannelData.java]: or even ```((msgLen - 1) / 4 + 1) * 4``` which is better
8/9/10/11 gives 8/12/12/12

**Comment by champtar** [comment]
I've reworked this PR to not break current API

**Comment by ibauersachs** [comment]
Apart from your good fix with the offset in decode, I don't quite see the use case for the other changes. Is there a problem if we always pad the messages?
Also:
- getDataLength() method still behaves differently (probably doesn't really matter)
- padTo4 as a separate method is overkill and pollutes the API, at the very least make it private

I'm still reluctant to change this at all. Can someone else please give some input? @bgrozev?

**Comment by champtar** [comment]
hi @ibauersachs,
We can always pad according to the RFC, but i prefer to be sure to keep back compat when possible.
getDataLength() is unchanged (it behaves exactly the same as before)
padTo4 is a bit unreadable, thus the separate function, but i will make it private.

I started this PR because length in getDataLength() was a dead store, and FindBugs spotted it.

**Comment by ibauersachs** [comment]
Oh, right, it was too late yesterday and I missed that getDataLength was returning the array length instead of the calculated value. So, okay, after making padto4 private I'm okay merging it.

**Comment by champtar** [comment]
padTo4 now private

**Pull request merged**